sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -172,41 +172,72 @@ def _create_pipeline(
 
     return pipeline_spec
 
+  def _create_pipeline_job(
+      self,
+      pipeline_spec: pipeline_spec_pb2.PipelineSpec,
+      pipeline_root: str,
+      pipeline_parameters: Optional[Mapping[str, Any]] = None,
+  ) -> pipeline_spec_pb2.PipelineJob:
+    """Creates the pipeline job spec object.
+
+    Args:
+      pipeline_spec: The pipeline spec object.
+      pipeline_root: The root of the pipeline outputs.
+      pipeline_parameters: The mapping from parameter names to values. Optional.
+
+    Returns:
+      A PipelineJob proto representing the compiled pipeline.
+    """
+    runtime_config = compiler_utils.build_runtime_config_spec(
+        pipeline_root=pipeline_root)
+    pipeline_job = pipeline_spec_pb2.PipelineJob(runtime_config=runtime_config)
+    pipeline_job.pipeline_spec.update(json_format.MessageToDict(pipeline_spec))
+
+    return pipeline_job
+
   def compile(self,
               pipeline_func: Callable[..., Any],
+              pipeline_root: str,
               output_path: str,
-              type_check: bool = True,
-              pipeline_name: str = None) -> None:
-    """Compile the given pipeline function into workflow yaml.
+              pipeline_name: Optional[str] = None,
+              pipeline_parameters: Optional[Mapping[str, Any]] = None,
+              type_check: bool = True) -> None:
+    """Compile the given pipeline function into pipeline job json.
 
     Args:
       pipeline_func: Pipeline function with @dsl.pipeline decorator.
+      pipeline_root: The root of the pipeline ouputs.
       output_path: The output pipeline spec .json file path. for example,
         "~/a.json"
       type_check: Whether to enable the type check or not, default: True.
       pipeline_name: The name of the pipeline. Optional.
+      pipeline_parameters: The mapping from parameter names to values. Optional.
```

Comment:
optional nit: make the order of docstring consistent with function signature

Reply:
Fixed.